File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/TA/TransactionalActivator.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Activation;
using Db4objects.Db4o.Internal;

namespace Db4objects.Db4o.TA
{
	/// <summary>
	/// An
	/// <see cref="Db4objects.Db4o.Activation.IActivator">Db4objects.Db4o.Activation.IActivator
	/// 	</see>
	/// implementation that activates an object on a specific
	/// transaction.
	/// </summary>
	/// <exclude></exclude>
	internal sealed class TransactionalActivator : IActivator
	{
		private readonly Transaction _transaction;

		private readonly ObjectReference _objectReference;

		public TransactionalActivator(Transaction transaction, ObjectReference objectReference
			)
		{
			_objectReference = objectReference;
			_transaction = transaction;
		}

		public void Activate(ActivationPurpose purpose)
		{
			_objectReference.ActivateOn(_transaction, purpose);
		}
	}
}

File: db4oj/core/src/com/db4o/ta/TransactionalActivator.java
package com.db4o.ta;

import com.db4o.activation.*;
import com.db4o.internal.*;

/**
 * An {@link Activator} implementation that activates an object on a specific
 * transaction.
 * 
 * @exclude
 */
final class TransactionalActivator implements Activator {
	
	private final Transaction _transaction;
	private final ObjectReference _objectReference;
	
	public TransactionalActivator(Transaction transaction, ObjectReference objectReference) {
		_objectReference = objectReference;
		_transaction = transaction;
	}

	public void activate(ActivationPurpose purpose) {
		_objectReference.activateOn(_transaction, purpose);
	}
}